vasiliy/src/main/java/data_structures/MyLinkedList.java

```diff
@@ -1,6 +1,207 @@
-public class MyLinkedList {
+public class MyLinkedList implements MyList {
     private Node head;
     private Node tail;
+    private int size;
+
+    public MyLinkedList(Object[] array) {
+        for (Object obj : array) {
+            this.add(obj);
+        }
+    }
+
+    private boolean checkIndex(int index) {
+        return index >= 0 && index < size;
+    }
+
+    @Override
+    public void clear() {
+        tail = null;
+        head = null;
+        size = 0;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    @Override
+    public boolean add(Object o) {
+        if (isEmpty()) {
+            head = new Node(o);
+            tail = head;
+        } else {
+            Node node = new Node(o, null, tail);
+            tail.setNext(node);
+            tail = node;
+        }
+        size++;
+        return true;
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        return checkIndex(indexOf(o));
+    }
+
+    @Override
+    public boolean add(int index, Object o) {
+        if (!checkIndex(index)) {
+            return false;
+        }
+        Node objectNode = new Node(o);
+        Node node;
+        if (index == size - 1) {
+            node = tail;
+        } else {
+            node = findNode(index);
+        }
+        node.setPrevious(objectNode);
+        objectNode.setNext(node);
+        if (index != 0) {
+            objectNode.setPrevious(node.getPrevious());
+            node.getPrevious().setNext(objectNode);
+        } else {
+            head = objectNode;
+        }
+        size++;
+        return true;
+    }
+
+    private Node findNode(int index) {
+        if (!checkIndex(index) || isEmpty()) {
+            return null;
+        }
+        if (index == size - 1) {
+            return tail;
+        }
+        Node node = head;
+        for (int i = 0; i < index; i++) {
+            node = node.getNext();
+        }
+        return node;
+    }
+
+    @Override
+    public Object get(int index) {
+        if (!checkIndex(index)) {
+            return null;
+        }
+        return findNode(index).getValue();
+    }
+
+    private void removeNode(Node node) {
+        if (size == 1) {
+            clear();
+            return;
+        }
+        if (node == head) {
+            node.getNext().setPrevious(null);
+            head = node.getNext();
+            node.setNext(null);
+            return;
+        }
+        if (node == head) {
+            node.getPrevious().setNext(null);
+            tail = node.getPrevious();
+            node.setPrevious(null);
+            return;
+        }
+        node.getPrevious().setNext(node.getNext());
+        node.getNext().setPrevious(node.getPrevious());
+        node.setNext(null);
+        node.setPrevious(null);
+        size--;
+    }
+
+    @Override
+    public Object remove(int index) {
+        if (!checkIndex(index)) {
```

Comment:
You will duplicate this check in method:
private Node findNode(int index) {
        if (!checkIndex(index) || isEmpty()) {
            return null;
        }